### PR TITLE
[AMDGPU] Rename COV module flag to amdhsa_code_object_version

### PR DESCRIPTION
**Pull request by epilk**
The previous name 'amdgpu_code_object_version', was misleading since this is really a property of the HSA OS. The new spelling also matches the asm directive I added in bc82cfb.

See comment thread here: https://github.com/llvm/llvm-project/pull/76267#discussion_r1445578615

### CODE REVIEW

**Comment by epilk** [comment]
@tstellar there was some uncertainty about whether to leave the spelling as-is since mesa used to following HSA code object versions (but doesn't anymore). Would you have any objections to this change?

**Review by arsenm — APPROVED** [review]

**Comment by saiislam** [comment]
FYI @DominikAdamski, in case anything needs to change in the flang driver.

File: llvm/test/CodeGen/AMDGPU/GlobalISel/crash-stack-address-O0.ll [hunk 1]
```diff
@@ -25,4 +25,4 @@ entry:
 }
 
 !llvm.module.flags = !{!0}
-!0 = !{i32 1, !"amdgpu_code_object_version", i32 500}
+!0 = !{i32 1, !"amdhsa_code_object_version", i32 500}
```
Comment (saiislam) [inline llvm/test/CodeGen/AMDGPU/GlobalISel/crash-stack-address-O0.ll:28]: Can we remove explicit mention of COV5 module flag in all these test files given that the current default is COV5?
Or, may be a separate patch would be better for that.

Comment (arsenm) [reply llvm/test/CodeGen/AMDGPU/GlobalISel/crash-stack-address-O0.ll:28]: Separate would be better

Comment (epilk) [reply llvm/test/CodeGen/AMDGPU/GlobalISel/crash-stack-address-O0.ll:28]: Sure, I'll make a PR for that too after this lands.
